php-checks/src/main/java/org/sonar/php/checks/formatting/FunctionSpacingCheck.java

```diff
@@ -19,85 +19,130 @@
+    super.visitMethodDeclaration(tree);
+  }
+
+  @Override
+  public void visitFunctionDeclaration(FunctionDeclarationTree tree) {
+    checkSpaceAfterFunctionName(tree.name().token(), tree.parameters().openParenthesisToken());
+    super.visitFunctionDeclaration(tree);
+  }
+
+  @Override
+  public void visitFunctionExpression(FunctionExpressionTree tree) {
+    checkClosureSpacing(tree.functionToken(), tree.parameters(),tree.lexicalVars());
+    super.visitFunctionExpression(tree);
+  }
 
-    AstNode lexicalVars = node.getFirstChild(PHPGrammar.LEXICAL_VARS);
-    if (lexicalVars != null && !isSpaceAround(lexicalVars.getFirstChild(PHPKeyword.USE), 1 /* space before */, 1 /* space after */)) {
-      formattingCheck.reportIssue("Put exactly one space before and after the \"use\" keyword.", lexicalVars);
+  private void checkClosureSpacing(SyntaxToken functionKeyword, ParameterListTree parameters, @Nullable LexicalVariablesTree lexicalVars) {
+    if (check.isClosureSpacing) {
+
+      if (TokenUtils.getNbSpaceBetween(functionKeyword, parameters.openParenthesisToken()) != 1) {
+        check.reportIssue(CLOSURE_SPACES_MESSAGE, functionKeyword);
+      }
+
+      if (lexicalVars != null) {
+        int spaceBeforeUse = TokenUtils.getNbSpaceBetween(parameters.closeParenthesisToken(), lexicalVars.useToken());
+        int spaceAfterUse = TokenUtils.getNbSpaceBetween(lexicalVars.useToken(), lexicalVars.openParenthesisToken());
+
+        if (spaceBeforeUse != 1 || spaceAfterUse != 1) {
+          check.reportIssue(CLOSURE_LEXICAL_VARS_SPACES_MESSAGE, lexicalVars);
+        }
+      }
     }
   }
 
   /**
    * Check there is not space between a function's name and the opening parenthesis.
    */
-  private void checkSpaceAfterFunctionName(FormattingStandardCheck formattingCheck, AstNode node) {
-    Token lParenToken = node.getFirstChild(PHPPunctuator.LPARENTHESIS).getToken();
-    Token funcNameToken = node.is(PHPGrammar.FUNCTION_CALL_PARAMETER_LIST) ?
-      node.getPreviousAstNode().getLastToken() : node.getFirstChild(PHPGrammar.IDENTIFIER).getToken();
-
-    if (getNbSpaceBetween(funcNameToken, lParenToken) != 0) {
-      formattingCheck.reportIssue("Remove all space between the method name \"" + funcNameToken.getOriginalValue() + "\" and the opening parenthesis.", node);
+  private void checkSpaceAfterFunctionName(SyntaxToken functionName, SyntaxToken openParenthesis) {
+    if (check.isNoSpaceAfterMethodName && TokenUtils.getNbSpaceBetween(functionName, openParenthesis) != 0) {
+      check.reportIssue(String.format(METHOD_NAME_SPACES_MESSAGE, functionName.text()), functionName);
     }
   }
 
   /**
    * Check space around the arguments' comma.
    */
-  private void checkSpaceForComma(FormattingStandardCheck formattingCheck, AstNode node) {
+  private void checkSpaceForComma(TokenVisitor tokenVisitor, List<SyntaxToken> commas, Tree tree) {
+    if (check.isOneSpaceAfterComma) {
+
     int msgIndex = -1;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 7. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)